arm-cognitiveservices/2016-02-01-preview/swagger/cognitiveservices.json

```diff
@@ -0,0 +1,691 @@
+        },
+        "properties": {
+          "$ref": "#/definitions/CognitiveServicesAccountPropertiesCreateParameters"
+        }
+      },
+      "required": [
+        "sku",
+        "kind",
+        "location",
+        "properties"
+      ],
+      "description": "The parameters to provide for the account."
+    },
+    "CognitiveServicesAccountUpdateParameters": {
+      "properties": {
+        "sku": {
+          "$ref": "#/definitions/Sku"
+        },
+        "tags": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "Gets or sets a list of key value pairs that describe the resource. These tags can be used in viewing and grouping this resource (across resource groups). A maximum of 15 tags can be provided for a resource. Each tag must have a key no greater than 128 characters and value no greater than 256 characters."
+        }
+      },
+      "description": "The parameters to provide for the account."
+    },
+    "Sku": {
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "Gets or sets the sku name. Required for account creation, optional for update.",
+          "enum": [
+            "F0",
+            "S0",
+            "S1",
+            "S2",
+            "S3",
+            "S4"
+          ],
+          "x-ms-enum": {
+            "name": "SkuName",
+            "modelAsString": false
+          }
+        },
+        "tier": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Gets the sku tier. This is based on the SKU name.",
+          "enum": [
+            "Free",
+            "Standard",
+            "Premium"
+          ],
+          "x-ms-enum": {
+            "name": "SkuTier",
+            "modelAsString": false
+          }
+        }
+      },
+      "required": [
+        "name"
+      ],
+      "description": "The SKU of the cognitive services account."
+    },
+    "CognitiveServicesAccountPropertiesCreateParameters": {
+      "description": "required empty proprties object"
+    },
+    "CognitiveServicesAccount": {
+      "properties": {
+        "etag": {
+          "type": "string",
+          "description": "Entity Tag"
+        },
+        "id": {
+          "type": "string",
+          "description": "The id of the created account"
+        },
+        "kind": {
+          "type": "string",
+          "description": "Type of cognitive service account."       
+        },
+        "location": {
+          "type": "string",
+          "description": "The location of the resource"
+        },
+        "name": {
+          "type": "string",
+          "description": "The name of the created account"
+        },
+        "properties": {
+          "$ref": "#/definitions/CognitiveServicesAccountProperties"
```

Comment:
You can flatten all the account properties one level up. This makes life easy for customers. They don't have to do properties.* for getting or setting something. Please take a look at this extension to get more info: https://github.com/Azure/autorest/blob/master/Documentation/swagger-extensions.md#x-ms-client-flatten.

All the other RPs compute, storage, network, etc. are extensively using this. It is pretty much a standard practice.